src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let currentState = Object.assign({}, state);
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'clear') {
+      currentState = {};
+      history.push(currentState);
+    }
+
+    if (actions[i].type === 'addProperties') {
+      const add = actions[i].extraData;
```

Comment:
[NAMING] - use proper names for object copy. Instead of naming the copy 'add', you can name it something more descriptive like 'propertiesToAdd' to make it clear what the variable represents.